internal/model/globalyaml.go

```diff
@@ -0,0 +1,29 @@
+package model
+
+type YAMLManifest struct {
```

Comment:
maybe assert it fulfills the interface so the code is easier to scan?

Reply:
ah I just mean `var _ WatchableManifest = YAMLManifest{}`

Reply:
ah, that seems superfluous to me when the interface is defined right next to the implementation. That's not explicitly against this principle, but you run the risk of violating it unnecessarily when you try to do that because you're tempted to export interfaces to add compile time checks throughout your codebase.

I don't feel strongly either way though, this method is just a thought technology I'm playing with.